crates/telio-firewall/src/firewall.rs

```diff
@@ -559,9 +610,7 @@ impl StatefullFirewall {
                 key,
                 connection_info
             );
-            if connection_info.is_remote_initiated
-                && !whitelist.is_port_whitelisted(peer, key.local_port)
-            {
+            if connection_info.is_remote_initiated && !is_port_whitelisted(key.link.local_port) {
                 telio_log_trace!("Removing UDP conntrack entry {:?}", key);
                 udp_cache.remove(&key);
```

Comment:
This part seems to have depended on the fact, that for whitelisted peers this functions was not called previously. As it removes the `is_remote_initiated` immediately. This has changed right now, so we need to check whitelist and remove entry from udp_cache only if both are true:
* peer is not whitelisted
* peer is talking over not whitelisted port

Reply:
Unfortunately, it seems like I have missed this during review on TCP too. Could we:
* Add a unit or integration test where `record_whitelisted == true` and some inbound connections (both, for TCP and UDP) are initiated to check if they can succesfully communicate past first response packet. (e.g. one outgoing packet and later two/three response packets).
* Fix inbound packet handling to also check whether peer is whitelisted (not just port) when doing cache removals

Reply:
Ok, let me fix these

Reply:
@Jauler actually I think the current implementation ticks all boxes. Can you take a look again? if I'm missing something? Note that `is_port_whitelisted()` is a closure which for whitelisted peer always returns `true`

Reply:
Oh yeah, you are right.

But in that case, maybe lets rename the closure? Generally this closure now checks whether the connection is whitelisted (and this means, that if `true` is returned it might be not port-whitelisted, but simply all incoming connections allowed). So maybe lets call it `is_connection_whitelisted` hopefully to avoid some misunderstandings.

Reply:
Done